### PR TITLE
Add result type and spawn keyword to tree-sitter grammar

### PR DESCRIPTION
**Pull request by SirZenith**
Adding result type and result type propagator definition to tree-sitter-v.

### CODE REVIEW

**Comment by SirZenith** [comment]
Also add `spawn` to keyword list.

File: server/tests/completion_test.v [hunk 1]
```diff
@@ -321,6 +321,12 @@ const completion_results = {
 		},
 	]
 	'import_symbols.vv':                     [
+		lsp.CompletionItem {
+			label: 'connect'
+			kind: .function
+			detail: 'pub fn pg.connect(config pg.Config) !pg.DB'
+			insert_text: 'connect'
+		},
```
Comment (spytheman) [inline server/tests/completion_test.v:329]: remove the commented lines from 348 to 354, since they should be working now

File: tree_sitter_v/node_types.v [hunk 2]
```diff
@@ -186,7 +187,7 @@ const supertype__expression_nodes = merge(supertype__expression_with_blocks_node
 	.type_cast_expression,
 	.type_initializer,
 	.type_selector_expression,
-	.unary_expression,
+	.unary_expression
```
Comment (spytheman) [inline tree_sitter_v/node_types.v:190]: It is better to have the ending `,` so that diffs are minimal when re-arranging/appending new items.
I think vfmt also adds the ending `,`, so you can just `v fmt -w tree_sitter_v/node_types.v` .

File: tree_sitter_v/node_types.v [hunk 3]
```diff
@@ -249,28 +251,29 @@ const supertype__top_level_declaration_nodes = [
...
 
 const supertype__type_nodes = merge(supertype__simple_type_nodes, [
 	NodeType.multi_return_type,
 	.option_type,
+	.result_type
 ])
 
 pub fn (typ NodeType) group() SuperType {
-	return if typ in tree_sitter_v.supertype__top_level_declaration_nodes {
+	return if typ in supertype__top_level_declaration_nodes {
```
Comment (spytheman) [inline tree_sitter_v/node_types.v:264]: Just run `v fmt -w tree_sitter_v/node_types.v` please - it will minimize the diff a lot.

File: tree_sitter_v/src/node-types.json [hunk 4]
```diff
@@ -3912,11 +3954,11 @@
   },
   {
     "type": "int_literal",
-    "named": true
+    "named": false
   },
   {
     "type": "int_literal",
-    "named": false
+    "named": true
```
Comment (spytheman) [inline tree_sitter_v/src/node-types.json:3961]: Why are these changed? These seem unrelated to the PR to me 🤔 .

**Review by spytheman — APPROVED** [review]
Good work.

File: tree_sitter_v/src/node-types.json (hunk 4 above)
Comment (SirZenith) [reply tree_sitter_v/src/node-types.json:3961]: This file is generated by tree-sitter, I think these tow objects simply exchanged their order.

File: tree_sitter_v/node_types.v (hunk 2 above)
Comment (SirZenith) [reply tree_sitter_v/node_types.v]: Done.

File: tree_sitter_v/node_types.v (hunk 3 above)
Comment (SirZenith) [reply tree_sitter_v/node_types.v]: Done.

File: server/tests/completion_test.v (hunk 1 above)
Comment (SirZenith) [reply server/tests/completion_test.v]: Uncommented that item, and remove new item added by me.
